jaeger-micrometer/src/test/java/io/jaegertracing/micrometer/MicrometerTest.java

```diff
@@ -85,4 +120,83 @@ public void testTimer() {
     assertThat(registry.get("jaeger:timed_operation").timer().totalTime(TimeUnit.MICROSECONDS), IsEqual.equalTo(100d));
     assertTrue(prometheusRegistry.scrape().contains("jaeger:timed_operation_seconds"));
   }
+
+
+  @Test
+  public void testExposedMetrics() {
+    Configuration configuration = new Configuration("exposedmetrics");
+    final Tracer tracer = configuration
+            .getTracerBuilder()
+            .withMetrics(metrics)
+            .build();
+
+    // This is a gauge, so it needs to be non-zero to come back from prometheus
+    metrics.reporterQueueLength.update(1);
+
+    List<Meter> meters = new ArrayList<>(prometheusRegistry.getMeters());
+    Map<String, Long> metricCounts = meters
+            .stream()
+            .collect(groupingBy(m -> m.getId().getName(), counting()));
+
+    assertEquals("Wrong number of metrics collected", expectedMetricCounts.size(), metricCounts.keySet().size());
+
+    for (String name : metricCounts.keySet()) {
+      assertTrue("Unexpected metric " +  name, expectedMetricCounts.containsKey(name));
+    }
+
+    for (String metricName : expectedMetricCounts.keySet()) {
+      assertTrue("Did not find metric " + metricName, metricCounts.containsKey(metricName));
+      assertEquals("Wrong count for " +  metricName, expectedMetricCounts.get(metricName),
+              metricCounts.get(metricName));
+    }
+
+    tracer.close();
+  }
+
+  @Test
+  public void validateMetricCounts() throws InterruptedException {
+    Sampler constantSampler = new ConstSampler(true);
+    Configuration configuration = new Configuration("validateMetricCounts");
+    Tracer tracer = configuration
+            .getTracerBuilder()
+            .withSampler(constantSampler)
+            .withMetrics(metrics)
+            .build();
+
+    createSomeSpans(tracer);
+    tracer.close();
+
+    double finishedSpans = registry.get("jaeger:finished_spans")
+            .counter()
+            .count();
+
+    double startedSpans = registry.get("jaeger:started_spans")
+            .tag("sampled", "y")
+            .counter()
+            .count();
+
+    double traces = registry.get("jaeger:traces")
+            .tag("sampled", "y")
+            .tag("state", "started")
+            .counter()
+            .count();
+
+    assertEquals("Wrong number of finishedSpans", 4.0, finishedSpans, assertDelta);
+    assertEquals("Wrong number of startedSpans", 10.0, startedSpans, assertDelta);
+    assertEquals("Wrong number of traces", 10.0, traces, assertDelta);
+  }
+
+  private void createSomeSpans(Tracer tracer) {
+    for (int i = 0; i < 10; i++) {
+      Span span = tracer.buildSpan("metricstest")
+              .withTag("foo", "bar" + i)
+              .start();
+
+      if (i % 3 == 0) {
```

Comment:
Can you add a comment explaining what this is for?

Reply:
I'm not quite sure I understand what is going on here: are you finishing only every third span? Would it cause any resource leaks?